addons/account/tests/test_account_move_out_invoice.py

```diff
@@ -3482,3 +3482,65 @@ def test_out_invoice_depreciated_account(self):
         self.product_a.property_account_income_id.deprecated = True
         with self.assertRaises(UserError), self.cr.savepoint():
             move.action_post()
+
+    @freeze_time('2019-01-01')
+    def test_date_reversal_exchange_move(self):
+        """
+        Test the date of the reversal of an exchange move created when unreconciling a payment made in the past.
+        It should be the last day of the month of the exchange move date if sequence is incremented by month,
+        and the last day of the year of the exchange move date if sequence is incremented by year.
+        """
+        def assert_reversal_date(format_incrementor, invoice_date, payment_date, expected_date):
+            invoice = self.init_invoice(move_type='out_invoice', partner=self.partner_a, invoice_date=invoice_date, post=True, amounts=[750.0], currency=self.currency_data['currency'])
+
+            new_exchange_journal = self.env['account.journal'].create({
+                'name': 'Exchange Journal for %s' % invoice.name,
+                'code': 'EXCH%s' % invoice.name,
+                'type': 'general',
+                'company_id': self.env.company.id,
+            })
+
+            # Need a first move in the new journal to initiate the sequence with a right incrementor, depending on the wanted format
+            first_exch_move = self.env['account.move'].create({
```

Comment:
> Also, I'm wondering: How is the user supposed to initialize the sequence by month for exchange journal?

Either by changing the sequence of the first move of the journal before posting it, or by resequencing an already posted move, which is not convenient in our use case, because the function that retrieve the last sequence is looking for a move posted before the exchange move (in our case it will be False because the exchange move is the only one in the exchange journal).

Reply:
Ok, so there isn't any "nice" way to setup that

Reply:
Not that I know of